polar-core/src/data_filtering.rs

```diff
@@ -212,126 +211,103 @@ impl VarInfo {
             // coming through it's probably a bug in the simplifier.
-            _ => unsupported_op_error(Operation {
+            _ => df_unsupported_op(Operation {
                 operator: Operator::Unify,
                 args: vec![left.clone(), right.clone()],
-            }),
+            })?,
         }
+        Ok(self)
     }
 
-    fn do_neq(mut self, left: &Term, right: &Term) -> Result<Self> {
+    fn do_neq(mut self, left: &Term, right: &Term) -> PolarResult<Self> {
         match (self.undot(left), self.undot(right)) {
-            (Value::Variable(l), Value::Variable(r)) => {
-                self.uncycles.push((l, r));
-                Ok(self)
-            }
+            (Value::Variable(l), Value::Variable(r)) => self.uncycles.push((l, r)),
             (Value::Variable(l), _) => {
                 let r = self.symbolize(right);
-                self.uncycles.push((l, r));
-                Ok(self)
+                self.uncycles.push((l, r))
             }
             (_, Value::Variable(r)) => {
                 let l = self.symbolize(left);
-                self.uncycles.push((l, r));
-                Ok(self)
+                self.uncycles.push((l, r))
             }
-            _ => unsupported_op_error(Operation {
+            _ => df_unsupported_op(Operation {
                 operator: Operator::Neq,
                 args: vec![left.clone(), right.clone()],
-            }),
+            })?,
         }
+        Ok(self)
     }
 
-    fn do_in(mut self, left: &Term, right: &Term) -> Result<Self> {
+    fn do_in(mut self, left: &Term, right: &Term) -> PolarResult<Self> {
         match (self.undot(left), self.undot(right)) {
-            (Value::Variable(l), Value::Variable(r)) => {
-                self.in_relationships.push((l, r));
-                Ok(self)
-            }
-            (val, Value::Variable(var)) => {
-                self.contained_values.push((Term::from(val), var));
-                Ok(self)
-            }
-            _ => unsupported_op_error(Operation {
+            (Value::Variable(l), Value::Variable(r)) => self.in_relationships.push((l, r)),
+            (val, Value::Variable(var)) => self.contained_values.push((Term::from(val), var)),
+            _ => df_unsupported_op(Operation {
                 operator: Operator::In,
                 args: vec![left.clone(), right.clone()],
-            }),
+            })?,
         }
+        Ok(self)
     }
 
     /// Process an expression in the context of this VarInfo. Just does side effects.
-    fn process_exp(self, exp: &Operation) -> Result<Self> {
+    fn process_exp(self, exp: &Operation) -> PolarResult<Self> {
         use Operator::*;
         let args = &exp.args;
         match exp.operator {
             And => self.do_and(args),
-            Dot if args.len() == 2 => self.do_dot(&args[0], &args[1]),
+            Dot if args.len() == 2 => Ok(self.do_dot(&args[0], &args[1])),
```

Comment:
I changed some of the return values for functions in this file that were unnecessarily returning a `Result` despite being infallible. I believe this was done for consistency at the call site (e.g., to avoid having to wrap this method call in an `Ok(...)`), but I think it's better to reserve `Result<T, E>` as an indication of fallibility.

Not a hill I'm willing to die on, so fine reverting these changes if others have strong opinions. (cc: @gw000)